src/Digdir.Domain.Dialogporten.Application/Features/V1/ServiceOwner/Dialogs/Commands/Create/CreateDialogCommandValidator.cs

```diff
@@ -281,7 +282,13 @@ internal sealed class CreateDialogDialogActivityDtoValidator : AbstractValidator
             .SetValidator(actorValidator);
         RuleFor(x => x.Description)
             .NotEmpty()
-            .SetValidator(localizationsValidator);
+            .SetValidator(localizationsValidator)
+            .When(x => x.Type == DialogActivityType.Values.Information)
+            .WithMessage("Description is required when the type is '" + nameof(DialogActivityType.Values.Information) + "'.");
+        RuleFor(x => x.Description)
+            .Empty()
+            .When(x => x.Type != DialogActivityType.Values.Information)
+            .WithMessage("Description is only allowed when the type is '" + nameof(DialogActivityType.Values.Information) + "'.");
```

Comment:
```suggestion
            .WithMessage("Description is required when the type is '" + nameof(DialogActivityType.Values.Information) + "'.")
            .SetValidator(localizationsValidator)
            .When(x => x.Type == DialogActivityType.Values.Information);
        RuleFor(x => x.Description)
            .Empty()
            .WithMessage("Description is only allowed when the type is '" + nameof(DialogActivityType.Values.Information) + "'.")
            .When(x => x.Type != DialogActivityType.Values.Information);
```

Tested this locally, and I missed this detail on first viewing.
The WithMessage belongs to the NotEmpty, so needs to be right after. 

The other way we get the default message, without the extra info:
```json
    "errors": {
        "Activities[0].Description": [
            "'Description' must not be empty."
        ]
    },
 ```
 
 Also, the When clause should be at the end 😊